internal/etc/magicbuilder/pgp.go

```diff
@@ -8,11 +8,29 @@ import (
 	"github.com/quan-to/chevron/internal/keymagic"
 	"github.com/quan-to/chevron/internal/vaultManager"
 	"github.com/quan-to/chevron/pkg/interfaces"
+	"github.com/quan-to/chevron/pkg/models"
 	"github.com/quan-to/slog"
 )
 
+type DatabaseHandler interface {
+	GetUser(username string) (um *models.User, err error)
+	AddUserToken(ut models.UserToken) (string, error)
+	RemoveUserToken(token string) (err error)
+	GetUserToken(token string) (ut *models.UserToken, err error)
+	InvalidateUserTokens() (int, error)
+	AddUser(um models.User) (string, error)
+	UpdateUser(um models.User) error
+	AddGPGKey(key models.GPGKey) (string, bool, error)
+	FindGPGKeyByEmail(email string, pageStart, pageEnd int) ([]models.GPGKey, error)
+	FindGPGKeyByFingerPrint(fingerPrint string, pageStart, pageEnd int) ([]models.GPGKey, error)
+	FindGPGKeyByValue(value string, pageStart, pageEnd int) ([]models.GPGKey, error)
+	FindGPGKeyByName(name string, pageStart, pageEnd int) ([]models.GPGKey, error)
+	FetchGPGKeyByFingerprint(fingerprint string) (*models.GPGKey, error)
+	HealthCheck() error
+}
```

Comment:
By reading the code, it looks like that you only need a subset of this interface, I am right? The `DatabaseHandler` is used to be inject to the `kaymagic.MakeKeyRingManager`, which only uses the following subset:

```suggestion
type DatabaseHandler interface {
	AddGPGKey(key models.GPGKey) (string, bool, error)
	FindGPGKeyByEmail(email string, pageStart, pageEnd int) ([]models.GPGKey, error)
	FindGPGKeyByFingerPrint(fingerPrint string, pageStart, pageEnd int) ([]models.GPGKey, error)
	FindGPGKeyByValue(value string, pageStart, pageEnd int) ([]models.GPGKey, error)
	FindGPGKeyByName(name string, pageStart, pageEnd int) ([]models.GPGKey, error)
	FetchGPGKeyByFingerprint(fingerprint string) (*models.GPGKey, error)
}
```

In this case you could either apply the above suggestion or, as this package seems to be proxying the db connection, this one:
```suggestion
type DatabaseHandler keymagic.DatabaseHandler
```
The advantage of defining the interface in the user of that interface is that you do not need to declare all the methods, but only the subset you'll use. What do you think?